tslint.json

```diff
@@ -81,19 +81,19 @@
         // TODO: "no-unused-expression": true,
         // TODO: "no-use-before-declare": true,
         "no-with-statement": true,
-        // TODO: "promise-function-async": true,
+        // TODO:"promise-function-async": true,
         // TODO: "promise-must-complete": true,
         "radix": true,
         "react-this-binding-issue": true,
         "react-unused-props-and-state": true,
-        // TODO: "restrict-plus-operands": true,
+        "restrict-plus-operands": false, // changed
         "strict-boolean-expressions": [
```

Comment:
I get annoyed by `strict-boolean-expressions`, too, but this is one that I begrudgingly keep because it's saved me several times. Specifically, when I had something like this:
```
async function foo(): Promise<boolean> {

}

if (foo()) {

}
```

You'll notice I forgot the 'await' when calling 'foo' and tslint will catch that because it doesn't like having a promise in the boolean expression.

Reply:
Not ready to completely throw in the towel on this one, although I'll have to see if there are some reasonable compromises with the options.